manual/src/cmds/profile-perf.etex

```diff
@@ -0,0 +1,195 @@
++    1.09%  sudoku_norvig.e  sudoku_norvig.exe  [.] Stdlib.List$fold_left_380            
+\end{verbatim}
+Note this implementation clearly has some optimsation possible in `_pow_747` a recursive
+Power of 2 function and parts of the OCaml runtime `caml_modify` are present.
+
+Profiling data can also be visualised as Flame Graphs, which highlight the most frequent code paths in stack traces. The original scripts scripts \texttt{stackcollapse-perf.pl} and \texttt{flamegraph.pl} can be found at Brendan Gregg's \href{https://www.brendangregg.com/flamegraphs.html}{Flame Graphs} web page and work as follows:
+
+\begin{verbatim}
+git clone https://github.com/brendangregg/FlameGraph
+cd FlameGraph
+
+# Collect the results into perf.data
+perf record -F 99 --call-graph fp <YOUR_EXECUTABLE>
+perf script -i perf.data | ./stackcollapse-perf.pl out.folded
+flamegraph.pl out.folded > flamegraph.svg ## Create the FlameGraph svg
+\end{verbatim}
+
+Alternatively \href{https://github.com/jonhoo/inferno}{inferno} is a Rust port of the Flame Graphs tools which works in a similar way and is faster to process large perf files.
+
+\begin{verbatim}
+cargo install inferno
+
+# Collect the results into perf.data
+perf script -i perf.data | inferno-collapse-perf > stacks.folded
+cat stacks.folded | inferno-flamegraph > flamegraph.svg
+\end{verbatim}
+
+Some \texttt{perf} tools (e.g., \texttt{perf-report(1)} and \texttt{perf-annotate(1)}) use DWARF debugging information to associate symbols with source code locations, if you need these features, the program and it's dependencies need to be compiled with \texttt{-g} to include debugging information in the executable.
+
+Captured profile data can also be processed using \texttt{perf script} in various ways or with online tools like \href{https://www.speedscope.app}{speedscope.app} and \href{https://profiler.firefox.com/}{profiler.firefox.com}, or any other tool that supports \texttt{perf}-formatted data.
+
+\section{s:ocamlperf-conclusion}{Conclusion}
+
+For CPU profiling of native code, standard tools such as \texttt{perf}, eBPF, DTrace, or Instruments (on macOS) are recommended. Compiling with frame pointers enabled is often necessary for these tools to work effectively. Profiling with \texttt{gprof} is no longer supported.
+
+Enabling frame pointers can impact performance on certain architectures (up to 10\% performance cost on x86_64 has been measured). Users of this feature are encouraged to benchmark their own applications to assess this impact.
+
+\section{s:ocamlperf-glossary}{Glossary}
+
+The following terminology is used in this chapter of the manual.
+
+\begin{itemize}
+\item[{\bf Call graph}] The chain of function calls that have lead to the current function (also referred to as a call stack).
+\item[{\bf Unwinding}] The process of restoring the program's state to how it was before some function(s) were called and possibly giving a profiler or debugger much more information. (also called stack unwinding).
```

Comment:
Nitpick: for me "unwinding" in this context is simply walking up the call stack (and being able to reconstruct its structure and content), it does not imply "restoring the program state" in any way.

Reply:
Fair point, I'll be more specific for this page. 
I originally wrote it in the context of source debuggers where unwinding does imply reconstructing previous frames and state. :-)